tests/routes/case.test.js

```diff
@@ -137,6 +168,122 @@ describe('/cases route', () => {
       });
     });
+    describe('With multiple records', () => {
+      const CASE_SAMPLE_SIZE = 10;
+      const createdCasesAndContacts = [];
+      const accounts = ['ACCOUNT_SID_1', 'ACCOUNT_SID_2'];
+      const helplines = ['helpline-1', 'helpline-2', 'helpline-3'];
+      beforeAll(async () => {
+        for (let i = 0; i < CASE_SAMPLE_SIZE; i += 1) {
+          const createdCase = await Case.create(
+            {
+              ...case1,
+              accountSid: accounts[i % accounts.length],
+              helpline: helplines[i % helplines.length],
+            },
+            options,
+          );
+          const createdContact = await Contact.create(
+            fillNameAndPhone({
+              ...contact1,
+              accountSid: accounts[i % accounts.length],
+              helpline: helplines[i % helplines.length],
+            }),
+            options,
+          );
+          createdContact.caseId = createdCase.id;
+          await createdContact.save(options);
+          createdCasesAndContacts.push({
+            contact: createdContact,
+            case: createdCase,
+          });
+        }
+      });
+
+      afterAll(async () => {
+        return Promise.all([
+          Case.destroy({ where: { id: createdCasesAndContacts.map(ccc => ccc.case.id) } }),
+          Contact.destroy({ where: { id: createdCasesAndContacts.map(ccc => ccc.contact.id) } }),
+        ]);
+      });
+
+      // eslint-disable-next-line jest/expect-expect
+      each([
+        {
+          description:
+            'should return all cases for account when no helpline, limit or offset is specified',
+          listRoute: `/v0/accounts/${accounts[0]}/cases`,
+          expectedCasesAndContacts: () =>
+            createdCasesAndContacts
+              .filter(ccc => ccc.case.dataValues.accountSid === accounts[0])
+              .sort((ccc1, ccc2) => ccc2.case.id - ccc1.case.id),
+          expectedCount: 5,
+        },
+        {
+          description: 'should return all cases for account & helpline when helpline is specified',
+          listRoute: `/v0/accounts/${accounts[0]}/cases?helpline=${helplines[1]}`,
+          expectedCasesAndContacts: () =>
+            createdCasesAndContacts
+              .filter(
+                ccc =>
+                  ccc.case.dataValues.accountSid === accounts[0] &&
+                  ccc.case.helpline === helplines[1],
+              )
+              .sort((ccc1, ccc2) => ccc2.case.id - ccc1.case.id),
+          expectedCount: 1,
+        },
+        {
+          description: 'should return first X cases when limit X is specified',
+          listRoute: `/v0/accounts/${accounts[0]}/cases?limit=3`,
+          expectedCasesAndContacts: () =>
+            createdCasesAndContacts
+              .filter(ccc => ccc.case.dataValues.accountSid === accounts[0])
+              .sort((ccc1, ccc2) => ccc2.case.id - ccc1.case.id)
+              .slice(0, 3),
+          expectedCount: 5,
```

Comment:
I'm a litle confused with `expectedCount`, shouldn't it be `3`?

Reply:
The `expectedCount` refers to the `totalCount` for the whole set of valid records before LIMIT & OFFSET are applied, I will rename it to make that clear

Reply:
oh, ok! Thanks for renaming it, much clearer now